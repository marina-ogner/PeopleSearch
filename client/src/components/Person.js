import React from "react";

class Person extends React.Component {
    render() {
        return (
            <div>
                {this.props.inf.length > 0 && <div className="grey">
                    {this.props.num && <p> Result #: {this.props.num} </p>}
                    {this.props.inf[0] && <p> First Name: {this.props.inf[0]} </p>}
                    {this.props.inf[1] && <p> Last Name: {this.props.inf[1]} </p>}
                    {this.props.inf[2] && <p> Address: {this.props.inf[2]} </p>}
                    {this.props.inf[3] && <p> City: {this.props.inf[3]} </p>}
                    {this.props.inf[4] && <p> State: {this.props.inf[4]} </p>}
                </div>}
            </div>

        );
    }
}
export default Person;