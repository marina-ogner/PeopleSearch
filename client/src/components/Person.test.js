import React from 'react';
import Person from './Person';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

test('Person component should render as expected', () => {
    const inf = ["MARK", "PRENTICE", "2413 W TEXAS ST", "MOSES LAKE", "WA"];
    const component = shallow(<Person inf={inf} num={1} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(6);
    expect(tree).toMatchSnapshot();
});
test('Person component should render with undefined value in this.props.inf', () => {
    const inf = ["MARK", undefined, "2413 W TEXAS ST", "MOSES LAKE", "WA"];
    const component = shallow(<Person inf={inf} num={1} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(5);
    expect(tree).toMatchSnapshot();
});
test('Person component should render with this.props.num==undefined', () => {
    const inf = ["MARK", "PRENTICE", "2413 W TEXAS ST", "MOSES LAKE", "WA"];
    const component = shallow(<Person inf={inf} num={undefined} />);
    const tree = toJson(component);
    expect(tree.children[0].children.length).toBe(5);
    expect(tree).toMatchSnapshot();
});
test('Person component should not render the inf inside of the div', () => {
    const inf = [];
    const component = shallow(<Person inf={inf} num={1} />);
    const tree = toJson(component);
    expect(tree.children).toBe(null);
    expect(tree).toMatchSnapshot();
});