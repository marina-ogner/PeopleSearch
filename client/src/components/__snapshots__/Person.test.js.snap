// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person component should not render the inf inside of the div 1`] = `<div />`;

exports[`Person component should render as expected 1`] = `
<div>
  <div
    className="grey"
  >
    <p>
       Result #: 
      1
       
    </p>
    <p>
       First Name: 
      MARK
       
    </p>
    <p>
       Last Name: 
      PRENTICE
       
    </p>
    <p>
       Address: 
      2413 W TEXAS ST
       
    </p>
    <p>
       City: 
      MOSES LAKE
       
    </p>
    <p>
       State: 
      WA
       
    </p>
  </div>
</div>
`;

exports[`Person component should render with this.props.num==undefined 1`] = `
<div>
  <div
    className="grey"
  >
    <p>
       First Name: 
      MARK
       
    </p>
    <p>
       Last Name: 
      PRENTICE
       
    </p>
    <p>
       Address: 
      2413 W TEXAS ST
       
    </p>
    <p>
       City: 
      MOSES LAKE
       
    </p>
    <p>
       State: 
      WA
       
    </p>
  </div>
</div>
`;

exports[`Person component should render with undefined value in this.props.inf 1`] = `
<div>
  <div
    className="grey"
  >
    <p>
       Result #: 
      1
       
    </p>
    <p>
       First Name: 
      MARK
       
    </p>
    <p>
       Address: 
      2413 W TEXAS ST
       
    </p>
    <p>
       City: 
      MOSES LAKE
       
    </p>
    <p>
       State: 
      WA
       
    </p>
  </div>
</div>
`;
